Make Noise seedable

DGruenwald opened the pull request:
Changed Math.random() to the Xorshift random number algorithm and added ability to set the seed, so, if needed, you can always get the same noise pattern.

This is my very first contribution to a project. If I did something wrong, please tell me!

DGruenwald committed:
Make Noise seedable

Changed Math.random() to the Xorshift random number algorithm and added ability to set the seed.

mrdoob commented:
I think it would be nicer if it was just `params.seed`. Isn't there a noise algorithm that takes only one value? How about [PRNG](https://github.com/mrdoob/three.js/blob/master/examples/js/PRNG.js)?

``` javascript
var value = (param.seed * 16807) % 2147483647;
```

(Sorry for the delay btw!)

DGruenwald commented:
Yes, that does look much better. I don't know what I was thinking...

DGruenwald committed:
Replace xorshift with a linear congruential generator

Using "Math.abs(value)" when setting the seed since most negative numbers result in a black image.

DGruenwald committed:
Make resulting number absolute instead of the seed

I realized why a negative seed made just a black image. It was because the resulting random numbers were negative.

mrdoob commented:
Sweet! Many thanks!
